### DISCUSSION TITLE
Can't write at memory 0 ?

### DISCUSSION THREAD

**Discussion by JMauperin** [gijzelaerr/python-snap7#356] [Q&A] [answered]
Hi everyone !

I'm not very accustomed to working with bytes, so if someone can help me understand my issue, I'll be really glad.

I created a fake s7 server (thanks for the server class, so useful !!) and I can read the data from the datablock with a client. But I tried to put a DWORD in a db before byte_index 6 and when I read it, I have a completly different value. Same thing when I try with a REAL before byte_index 8 in a db.

Obviously, a DWORD is 6 bytes and a REAL 8 so it is related. But I'd like to know why.

And is it the same with a real s7 plc ? Because I know someone who wants to put data on the first bytes of many datablocks...

Thank you to everyone who can help me understand.

**Comment by swamper123** [comment]
Hey ho!

A DWORD  (double word) contains 2 WORD and each WORD contains 2 BYTES -- 8 Bytes in sum. ;)

Could this be the fault?

**Reply by JMauperin** [reply]
Possible but even so, why can't I write from the start of the bytearray ? There must be an explanation of this.
It's not an issue for me, I'm just doing a soft to read data. But I would have like to understand it.

**Reply by swamper123** [reply]
Could you provide us a  code example you used?

**Comment by gijzelaerr** [comment, marked as answer]
Closing due to age and inactivity. For reference: a DWORD is 4 bytes and a REAL is 4 bytes. Writing at byte offset 0 should work fine. If you experienced issues, it was likely due to incorrect byte offsets or sizes in the read/write calls.
